internal/dag/httpproxy_processor.go

```diff
@@ -127,6 +120,13 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 	}
 	pa.Vhost = host
 
+	// ensure root httpproxy lives in allowed namespace
```

Comment:
Add a comment explaining why the ordering of these checks is important?

Reply:
Good idea, thanks.